app/controllers/lti_deployments_controller.rb

```diff
@@ -28,75 +30,119 @@ def launch
       prompt: 'none',
       state: session.id # Binds this request to the LTI response
     }
-    referrer = URI(request.referer)
-
-    auth_request_uri = URI("#{referrer.scheme}://#{referrer.host}:#{referrer.port}#{self.class::LMS_REDIRECT_ENDPOINT}")
+    referrer_uri = URI(request.referer)
+    referrer_uri = referrer_uri.to_s.remove(referrer_uri.path) if referrer_uri.path
```

Comment:
Does remove always remove from the right end of the string? What if there's a query that occurs in the host name. I think it's safer to reconstruct the query from the left side instead. Something like:

```ruby
referer = URI(request.referer)
referer_host = "#{referer.scheme}://#{referer.host}"
referer_host_with_port = "#{referer_host}:#{referer.port}"
referer_host = referer_host_with_port if referer.to_s.start_with?(referer_host_with_port)
auth_request_uri = URI("#{referer_host}#{self.class::LMS_REDIRECT_ENDPOINT}")
```